### PR TITLE
Add support to custom type hint

### LINKED ISSUE #1002: Allow doctrine custom types to register their php class to makerbundle for generating Type Hint
Right now its hardcoded in https://github.com/symfony/maker-bundle/blob/f76abb1e4facd712a5cdbf35aff998e87c416801/src/Util/ClassSourceManipulator.php#L1116-L1163

It would be useful to allow a configuration or any other way to allow user code to extend this type hint mappings. Right now for any fields of doctrine custom types, maker bundle does not generate any type hint for getter and setter methods. Which is expected as `Doctrine\DBAL\Types\Type` does not have any way to convey php object's type. Its cumbersome to manually add type hint to getter and setter, and its pointless as when regenerating entity (make:entity --regenerate --overwrite) its wiped out anyways.

### PR DESCRIPTION
**Pull request by IndraGunawan**
Fix #1002

### CODE REVIEW

**Comment by IndraGunawan** [comment]
@weaverryan this PR was ready, i don't think the CI failure related to this PR, any suggestions?

**Comment by IndraGunawan** [comment]
@jrushlow this PR is ready for review

**Comment by vladdnepr** [comment]
up. cool to merge

File: tests/DependencyInjection/ConfigurationTest.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Indragunawan\ApiRateLimitBundle\Tests\DependencyInjection;
```
Comment (jrushlow) [inline tests/DependencyInjection/ConfigurationTest.php:12]: This should be in the `Symfony\Bundle\MakerBundle\Tests`... namespace

**Review by jrushlow — CHANGES_REQUESTED** [review]
Sorry for the delay @IndraGunawan - I didnt dig through this too deep as I believe there are quite a few changes that have taken place since you first created the PR. If you can rebase and get us green tests, ill get this reviewed && merged.. Thanks in advance!

**Pull request closed without merging**